File: Arrays/Two Number SUm/python/solution1.py

# O(n2) Time Complexity
# Using Double loop O(n2) Time Comlpexity

def twoNumberSum(array, targetSum):
	for i in range(0,len(array)):
		for j in range(i,len(array)):
			if i!=j:
				if array[i]+array[j]==targetSum:
					return [array[i],array[j]]
	return []

File: Arrays-Old/Two Number SUm/c++/solution1.cpp
#include <vector>
#include <algorithm>
using namespace std;

// O(n2) Time Complexity | O(1) Complexity

vector<int> twoNumberSum(vector<int> array, int targetSum) {
  // Write your code here.
	for(int i =0;i< array.size()-1;i++){
		int first = array[i];
		for (int j = i +1; j< array.size();j++) {
			int second = array[j];
			if(first + second == targetSum){
				return vector<int>{first,second};
			}
		}
	}
  return {};
}
